src/components/VmStatus/getVmStatus.js

```diff
@@ -156,17 +126,132 @@ const isWaitingForVmi = vm => {
   return NOT_HANDLED;
 };
 
-export const getVmStatusDetail = (vm, launcherPod, importerPods, migration) =>
-  isBeingMigrated(vm, migration) || // must be precceding isRunning() since vm.status.ready is true for a migrating VM
+const findCdiPod = (volume, cdiPods = []) =>
+  cdiPods.find(pod => {
+    const podName = getName(pod);
+    const dataVolumeName = volume.dataVolume.name;
+    // upload dataVolume pod
+    if (podName === `cdi-upload-${dataVolumeName}`) {
+      return true;
+    }
+    // import dataVolume pod
+    const importerPodName = `importer-${dataVolumeName}-`;
+    if (podName.startsWith(importerPodName) && podName.length === importerPodName.length + 5) {
```

Comment:
why 5? Better to put some explanation and/or make a variable out of it

Reply:
right, 5 is actually not really good. It was meant to be a suffix generated by kubernetes but It seems that length does not always have to be 5. Im looking at pod labels now, which is definitely better.